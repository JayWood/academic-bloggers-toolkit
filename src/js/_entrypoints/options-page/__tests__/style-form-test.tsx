import * as React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import StyleForm from '../style-form';

const setup = () => {
    const component = shallow(<StyleForm />);
    return {
        component,
    };
};

describe('<StyleForm />', () => {
    it('should match snapshots', () => {
        const { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();
    });
});
