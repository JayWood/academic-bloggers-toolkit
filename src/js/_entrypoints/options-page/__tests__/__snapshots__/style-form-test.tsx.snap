// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyleForm /> should match snapshots 1`] = `
<React.Fragment>
  <devtool
    position={
      Object {
        "left": 50,
        "top": 40,
      }
    }
  />
  <RadioGroup
    items={
      Array [
        Object {
          "label": "Predefined",
          "value": "predefined",
        },
        Object {
          "label": "Custom",
          "value": "custom",
        },
      ]
    }
    label="Citation Style Type"
    name="style_kind"
    onChange={[Function]}
    style={
      Object {
        "margin": "10px 0",
      }
    }
    value="predefined"
  />
  <input
    name="style_value"
    type="hidden"
    value="american-medical-association"
  />
  <input
    name="style_label"
    type="hidden"
    value="American Medical Association"
  />
  <StyleInput
    currentStyle={
      Store {
        "kind": "predefined",
        "label": "American Medical Association",
        "value": "american-medical-association",
      }
    }
    onSelected={[Function]}
    styles={
      Array [
        Object {
          "kind": "predefined",
          "label": "American Medical Association",
          "value": "american-medical-association",
        },
        Object {
          "kind": "custom",
          "label": "Test Custom Style",
          "value": "test-custom-style",
        },
      ]
    }
  />
</React.Fragment>
`;
